src/esp/nav/PathFinder.cpp

```diff
@@ -924,15 +924,30 @@ static float frand() {
 }
 
 vec3f PathFinder::Impl::getRandomNavigablePoint() {
-  dtPolyRef ref;
+  constexpr int MAX_TRIES = 1e3;
   constexpr float inf = std::numeric_limits<float>::infinity();
-  vec3f pt(inf, inf, inf);
-  dtStatus status =
-      navQuery_->findRandomPoint(filter_.get(), frand, &ref, pt.data());
-  if (!dtStatusSucceed(status)) {
+  if (getNavigableArea() <= 0.0)
+    throw std::runtime_error(
+        "NavMesh has no navigable area, this indicates an issue with the "
+        "NavMesh");
+
+  vec3f pt;
+
+  int i;
+  for (i = 0; i < MAX_TRIES; ++i) {
+    dtPolyRef ref;
+    dtStatus status =
+        navQuery_->findRandomPoint(filter_.get(), frand, &ref, pt.data());
+    if (dtStatusSucceed(status))
+      break;
+  }
+
+  if (i == MAX_TRIES) {
     LOG(ERROR) << "Failed to getRandomNavigablePoint";
+    return {inf, inf, inf};
```

Comment:
Maybe we should make this NaN for consistency. I know we had it as inf before though.